src/Countries/Australia.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Australia extends Country
+{
+    protected function __construct(
+        protected ?string $region = null,
+    ) {
+    }
+
+    public function countryCode(): string
+    {
+        return 'au';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'New Year\'s Day' => '01-01',
+            'Australia Day' => '01-26',
+            'Anzac Day' => '04-25',
+            'Christmas Day' => '12-25',
+            'Boxing Day' => '12-26',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = $this->easter($year)->setTimezone('Australia/Sydney');
+
+        return [
+            'Good Friday' => $easter->subDays(2),
+            'Easter Monday' => $easter->addDay(),
+            // https://en.wikipedia.org/wiki/Public_holidays_in_Australia
+            ...array_filter(match($this->region) {
+                'act' => [
+                    'Canberra Day' => CarbonImmutable::parse("second monday of march {$year}"),
+                    'Easter Saturday' => $easter->subDay(),
+                    'Easter Sunday' => $easter,
+                    'Reconciliation Day' => CarbonImmutable::create($year, 5, 27)->modify("monday"),
```

Comment:
@Nielsvanpach would you rather this PHPStan failure ignored or a ternary?

```php
'Reconciliation Day' => ($may27 = CarbonImmutable::create($year, 5, 27)) ? $may27->modify("monday") : null,
```

Reply:
I would suggest something like:

```php
Carbon::create($year, 5, 27)->modify('monday')->toImmutable()
```

Reply:
This gives two errors 😄 

```
  45     Cannot call method modify() on Carbon\Carbon|false.
  45     Cannot call method toImmutable() on Carbon\Carbon|false.
  ```

Reply:
Added to the phpstan baseline, there were others in there already so I assume that's the right course of action here.